### ISSUE TITLE
Iteration number count in Training log is off

### ISSUE THREAD

**Issue by fareeha** [apple/turicreate#2290]
In DC toolkit (maybe an issue in others too) the `iteration` count jumps:

For 10 epochs on data size of 200  i.e. 8 iterations:
(skips 5 and jumps to 6)
```
+------------------------+------------------------+------------------------+------------------------+
| Iteration              | Training Loss          | Training Accuracy      | Elapsed Time (seconds) |
+------------------------+------------------------+------------------------+------------------------+
| 1                      | 0.757                  | 0.495                  | 0.445                  |
+------------------------+------------------------+------------------------+------------------------+
| 2                      | 0.691                  | 0.540                  | 0.777                  |
+------------------------+------------------------+------------------------+------------------------+
| 3                      | 0.674                  | 0.640                  | 1.112                  |
+------------------------+------------------------+------------------------+------------------------+
| 4                      | 0.627                  | 0.780                  | 1.471                  |
+------------------------+------------------------+------------------------+------------------------+
| 6                      | 0.595                  | 0.860                  | 1.822                  |
+------------------------+------------------------+------------------------+------------------------+
| 7                      | 0.557                  | 0.905                  | 2.140                  |
+------------------------+------------------------+------------------------+------------------------+
| 8                      | 0.533                  | 0.915                  | 2.447                  |
+------------------------+------------------------+------------------------+------------------------+
| 9                      | 0.512                  | 0.925                  | 2.763                  |
+------------------------+------------------------+------------------------+------------------------+
```

For 20 epochs on data size of 200  i.e. 16 iterations:
(skips 5 and jumps to 6, skips 10 and jumps to 11, skips 14 and jumps to 15, skips 19 and jumps to 20)
```
+------------------------+------------------------+------------------------+------------------------+
| Iteration              | Training Loss          | Training Accuracy      | Elapsed Time (seconds) |
+------------------------+------------------------+------------------------+------------------------+
| 1                      | 0.757                  | 0.495                  | 0.405                  |
+------------------------+------------------------+------------------------+------------------------+
| 2                      | 0.691                  | 0.540                  | 0.713                  |
+------------------------+------------------------+------------------------+------------------------+
| 3                      | 0.674                  | 0.640                  | 1.020                  |
+------------------------+------------------------+------------------------+------------------------+
| 4                      | 0.627                  | 0.780                  | 1.340                  |
+------------------------+------------------------+------------------------+------------------------+
| 6                      | 0.595                  | 0.860                  | 1.652                  |
+------------------------+------------------------+------------------------+------------------------+
| 7                      | 0.557                  | 0.905                  | 1.963                  |
+------------------------+------------------------+------------------------+------------------------+
| 8                      | 0.533                  | 0.915                  | 2.277                  |
+------------------------+------------------------+------------------------+------------------------+
| 9                      | 0.516                  | 0.925                  | 2.635                  |
+------------------------+------------------------+------------------------+------------------------+
| 11                     | 0.495                  | 0.935                  | 2.940                  |
+------------------------+------------------------+------------------------+------------------------+
| 12                     | 0.475                  | 0.935                  | 3.244                  |
+------------------------+------------------------+------------------------+------------------------+
| 13                     | 0.464                  | 0.935                  | 3.562                  |
+------------------------+------------------------+------------------------+------------------------+
| 15                     | 0.453                  | 0.965                  | 3.875                  |
+------------------------+------------------------+------------------------+------------------------+
| 16                     | 0.433                  | 0.970                  | 4.181                  |
+------------------------+------------------------+------------------------+------------------------+
| 17                     | 0.425                  | 0.970                  | 4.497                  |
+------------------------+------------------------+------------------------+------------------------+
| 18                     | 0.417                  | 0.975                  | 4.823                  |
+------------------------+------------------------+------------------------+------------------------+
| 20                     | 0.409                  | 0.975                  | 5.046                  |
+------------------------+------------------------+------------------------+------------------------+
``` 

The data loader correctly works in order, just the counts printed are off.

**Comment by dhivyaaxim** [comment]
@znation  Please assign this ticket to me as I would like to try fixing it and contribute. Thanks in advance!

**Comment by TobyRoseman** [comment]
@dhivyaaxim - Thanks for your offer to help here. However this may not be a good one to do right now. TuriCreate is currently in the process of swapping out our current deep learning dependency (MXNet) for TensorFlow. So the code which generates this table will probably be changing a lot in the near future. Once we're done moving to TensorFlow this issue might even fix itself.

**Comment by TobyRoseman** [comment]
This issue was created before 6.0. So things may have changed, but it looks like the skipping of rows is by design.

The [Drawing Classifier uses `print_progress_row`](https://github.com/apple/turicreate/blob/269730670de0d674e921344432abf6f7dfe3c48c/src/toolkits/drawing_classifier/drawing_classifier.cpp#L601-L619) which skips rows, see [the doc strings](https://github.com/apple/turicreate/blob/af567ff5ba333f39ff9b7dac44e74a7b558c0da7/src/core/logging/table_printer/table_printer.hpp#L108-L113).

This does indeed look strange when only doing ten iterations. However when doing a lot more iterations (typical use case) it makes sense.

In investigating this issue, I noticed iterations one larger than a round number are getting printed (ex: iteration 501 rather than 500). I've fixed this in #3353.
